### PR TITLE
Fix graph.py to handle multiple output nodes

### PR DESCRIPTION
**Pull request by orionr**
Rebased version of https://github.com/lanpa/tensorboardX/pull/162

cc @lanpa @geof90

### CODE REVIEW

**Comment by orionr** [comment]
@lanpa any chance you could look at the CI failure? Much appreciated.

File: tensorboardX/pytorch_graph.py [hunk 1]
```diff
@@ -52,19 +55,18 @@ def parse(graph):
...
-            nodes.append({'name': uname, 'op': n.kind(),
-                          'inputs': inputs, 'attr': attrs})
+
+        for outputnode in iter(n.outputs()):
+            uname = outputnode.uniqueName()
+            if outputnode.type().kind() == 'TensorType':
+                outputsize = outputnode.type().sizes()
+                nodes.append({'name': uname,
+                              'op': n.kind(),
+                              'inputs': inputs,
```
Comment (geof90) [inline tensorboardX/pytorch_graph.py]: This way, multiple nodes will share the same underlying inputs array, so when we modify them below, we get the `KeyError` that the CI is failing with. The way to fix this is to initialize a seperate `inputs` array during each iteration of the loop.

**Comment by geof90** [comment]
@orionr Thanks for sending this out! BTW line 19 needs to be fixed as well.

**Comment by orionr** [comment]
Test failure seems unrelated. Not sure how to re-run, though.

**Comment by lanpa** [comment]
python 2.7 on CI was broken since
https://travis-ci.org/lanpa/tensorboardX/builds/448393574

The graph (AlexNet) looks weird on 
`pytorch-nightly           1.0.0.dev20181108 py3.6_cuda9.0.176_cudnn7.1.2_0`

What version are you using?

image

**Comment by orionr** [comment]
Using latest PyTorch master, but we didn't test on AlexNet. @geof90 looks like we need to debug a bit?

**Comment by lanpa** [comment]
Just confirmed that the plot failed since torch-nightly (1.0.0.dev20181103)

**Pull request merged**